Source/engine/render/automap_render.cpp

```diff
@@ -94,4 +96,103 @@ void DrawMapLineSteepSW(const Surface &out, Point from, int width, std::uint8_t
 	DrawMapLineSteep<DirectionX::WEST, DirectionY::SOUTH>(out, from, width, colorIndex);
 }
 
+void DrawMapEllipse(const Surface &out, Point from, int radius, uint8_t colorIndex)
+{
+	int a = radius;
+	int b = radius / 2;
+
+	int x = 0;
+	int y = b;
+
+	// Initial point
+	out.SetPixel({ from.x, from.y + b }, colorIndex);
+	out.SetPixel({ from.x, from.y - b }, colorIndex);
+
+	// Initialize the parameters
+	int p1 = (b * b) - (a * a * b) + (a * a) / 4;
+
+	// Region 1
+	while ((2 * b * b * x) < (2 * a * a * y)) {

```

Comment:
You don't need to multiply by 2 here. This comparison is the same as simply:

```suggestion
	while (b * b * x < a * a * y) {
```